mvm/machine/receiver.go

```diff
@@ -17,7 +17,10 @@ func (m *Machine) ProcessOutput(ctx context.Context, out *mtg.Output) {
 	}
 	switch op.Purpose {
 	case encoding.OperationPurposeAddProcess:
-		m.AddProcess(ctx, op.Process, op.Platform, op.Address, out, op.Extra)
+		ok := m.AddProcess(ctx, op.Process, op.Platform, op.Address, out, op.Extra)
+		if ok && op.Platform == ProcessPlatformEos {
+			m.WriteGroupEvent(op.Process, out, op.Extra)
```

Comment:
Is this only used to increase the nonce? I didn't see anything related to this kind of group event usage.

Reply:
Nope. The Eos engine will push an `addprocess` action to `mtg.xin` contract when receiving `addprocess` event, so `mtg.xin` contract can verify if a contract is a MVM contract in order to filter out invalid events.